test/libsolidity/SyntaxTest.cpp

```diff
@@ -46,15 +46,25 @@ SyntaxTest::SyntaxTest(
 	CommonSyntaxTest(_filename, _evmVersion),
 	m_minSeverity(_minSeverity)
 {
-	static std::set<std::string> const compileViaYulAllowedValues{"true", "false"};
-
 	auto const eofEnabled = solidity::test::CommonOptions::get().eofVersion().has_value();
 
-	m_compileViaYul = m_reader.stringSetting("compileViaYul", eofEnabled ? "true" : "false");
-	if (!util::contains(compileViaYulAllowedValues, m_compileViaYul))
-		BOOST_THROW_EXCEPTION(std::runtime_error("Invalid compileViaYul value: " + m_compileViaYul + "."));
+	m_compileViaYul = m_reader.enumSetting<CompileViaYul>(
+		"compileViaYul",
+		{
+			{toString(CompileViaYul::False), CompileViaYul::False},
+			{toString(CompileViaYul::True), CompileViaYul::True},
+			{toString(CompileViaYul::Also), CompileViaYul::Also},
+			{toString(CompileViaYul::onlyOnEOF), CompileViaYul::onlyOnEOF}
+		},
+		toString(CompileViaYul::onlyOnEOF)
+	);
+
+	solUnimplementedAssert(m_compileViaYul != CompileViaYul::Also);
```

Comment:
These should always include a message for the user (#15671):

```suggestion
	solUnimplementedAssert(m_compileViaYul != CompileViaYul::Also, "'compileViaYul: also' is not yet supported in syntax tests.");
```